misc/vtk/src/processors/vtktovolume.cpp

```diff
@@ -0,0 +1,406 @@
+            numTotalComps += dataArrays_.numArrayComponents[p];
+        }
+    }
+    if (!numTotalComps) {
+        LogWarn("No components selected.");
+        return;
+    }
+
+    const DataFormatBase *multichannelFormat =
+        DataFormatBase::get(format->getNumericType(), numTotalComps, format->getSize() * 8);
+
+    // Use point bounds
+    Matrix<4, float> modelMatrix(0);
+    double bounds[6];
+    vtkData->GetBounds(bounds);
+    for (size_t x = 0; x < 3; ++x) {
+        modelMatrix[x][x] = (bounds[x * 2 + 1] - bounds[x * 2]) / dimensions[x];
+        modelMatrix[3][x] = bounds[x * 2];
+    }
+    modelMatrix[3][3] = 1;
+
+    // Create new volume
+    auto volume = std::make_shared<Volume>(dimensions, multichannelFormat);
+    volume->setModelMatrix(modelMatrix);
+
+    // Create RAM volume
+    auto volRAM = createVolumeRAM(dimensions, multichannelFormat);
+    unsigned char *dataPtr = static_cast<unsigned char *>(volRAM->getData());
+
+    // Copy the data from the other volumes to the new multichannel volume
+    size_t numElements = dimensions.x * dimensions.y * dimensions.z;
+    for (size_t e = 0; e < numElements; e++) {
+        for (auto &arrayData : arrayDataPointers) {
+            memcpy(dataPtr, arrayData.ptr, arrayData.byteSize);
+            dataPtr += arrayData.byteSize;
+            arrayData.ptr += arrayData.byteSize;
+        }
+    }
+
+    volume->addRepresentation(volRAM);
+    double minVal(std::numeric_limits<double>::max()), maxVal(std::numeric_limits<double>::min());
+    for (size_t z = 0; z < dimensions.z; ++z)
+        for (size_t y = 0; y < dimensions.y; ++y)
+            for (size_t x = 0; x < dimensions.x; ++x) {
+                auto data = volRAM->getAsDVec4(size3_t(x, y, z));
+                for (size_t d = 0; d < numTotalComps; ++d) {
+                    minVal = std::min(data[d], minVal);
+                    maxVal = std::max(data[d], maxVal);
+                }
+            }
+
+    volume->dataMap_.valueRange = {minVal, maxVal};
+    volume->dataMap_.dataRange = {dataRange_.get().x, dataRange_.get().y};
+    outport_.setData(volume);
+}
+
+const DataFormatBase *VTKtoVolume::getFormatFromVtkId(int vtkTypeId, int bitSize) {
+    switch (vtkTypeId) {
+        case VTK_CHAR:
+        case VTK_SIGNED_CHAR:
+        case VTK_SHORT:
+        case VTK_INT:
+        case VTK_LONG:
+        case VTK_LONG_LONG:
+        case VTK___INT64:
+            return DataFormatBase::get(NumericType::SignedInteger, 1, bitSize);
+
+        case VTK_UNSIGNED_CHAR:
+        case VTK_UNSIGNED_SHORT:
+        case VTK_UNSIGNED_INT:
+        case VTK_UNSIGNED_LONG:
+        case VTK_UNSIGNED_LONG_LONG:
+        case VTK_UNSIGNED___INT64:
+        case VTK_ID_TYPE:
+            return DataFormatBase::get(NumericType::UnsignedInteger, 1, bitSize);
+
+        case VTK_FLOAT:
+        case VTK_DOUBLE:
+            return DataFormatBase::get(NumericType::Float, 1, bitSize);
+        default:
```

Comment:
This code should potentially be moved to some vtk utils. I imagine it being used quite often.

Reply:
That's true, I'll move it.